File: internal/daeconfig/loglevel_test.go
package daeconfig

import "testing"

func TestLogLevel(t *testing.T) {
	tests := []struct {
		name    string
		content string
		want    string
	}{
		{name: "多行", content: "global {\n  log_level: warn\n}\n", want: "warn"},
		{name: "紧凑写法", content: "global { log_level: debug }", want: "debug"},
		{name: "默认值", content: "global { lan_interface: auto }", want: "info"},
		{
			name: "跳过注释与字符串",
			content: `# global { log_level: error }
node { demo: 'global { log_level: trace }' }
/* global { log_level: warn } */
global { log_level: info }`,
			want: "info",
		},
		{name: "未知值", content: "global { log_level: verbose }", want: ""},
	}
	for _, test := range tests {
		t.Run(test.name, func(t *testing.T) {
			if got := LogLevel(test.content); got != test.want {
				t.Fatalf("LogLevel() = %q，期望 %q", got, test.want)
			}
		})
	}
}

File: web/src/utils/loglevel.test.ts
import { describe, expect, it } from 'vitest'
import { readDaeLogLevel, replaceDaeLogLevel } from './loglevel'

describe('dae 日志级别配置', () => {
  it('读取显式值与默认值', () => {
    expect(readDaeLogLevel('global {\n  log_level: warn\n}\n')).toBe('warn')
    expect(readDaeLogLevel('global {\n  lan_interface: auto\n}\n')).toBe('info')
  })

  it('拒绝歧义值', () => {
    expect(readDaeLogLevel('global {\n  log_level: info\n  log_level: debug\n}\n')).toBeNull()
    expect(readDaeLogLevel('global {\n  log_level: verbose\n}\n')).toBeNull()
  })

  it('只改目标声明并保留注释', () => {
    const original = 'global {\n  log_level: warn # 运维备注\n  lan_interface: auto\n}\n'
    expect(replaceDaeLogLevel(original, 'info')).toBe(
      'global {\n  log_level: info # 运维备注\n  lan_interface: auto\n}\n',
    )
  })
})
